### PR TITLE
add waypoints for commissions in dwarven mines and glacite.

### PR DESCRIPTION
**Pull request by olim88**
Shows labels in the world for areas in dwarven mines and glacite tunnels that are linked to an active commission

# examples
image
image

the glaciate waypoints  are just locations that were sent in dc by Fluboxer. 
the dwarven waypoints are from wiki

### CODE REVIEW

**Comment by viciscat** [comment]
Waypoints for the emissaries maybe? could be nice

File: src/main/resources/assets/skyblocker/lang/en_us.json [hunk 1]
```diff
@@ -397,6 +397,22 @@
   "skyblocker.config.mining.crystalsWaypoints.findInChat.@Tooltip": "When in crystal hollows read the chat to see if coordinates are sent and extract these to show as waypoint or on the map",
   "skyblocker.config.mining.crystalsWaypoints.shareFail": "Can only share waypoints you have found.",
 
+  "skyblocker.config.mining.commissionWaypoints": "Commission Waypoints",
```
Comment (AzureAaron) [inline src/main/resources/assets/skyblocker/lang/en_us.json:400]: Put these in alphabetical order with the rest of the translation strings

File: src/main/java/de/hysky/skyblocker/skyblock/dwarven/MiningLocationLabel.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+            return "Emissary";
+        }
+
+        @Override
+        public int getColor() {
+            return 0xffffff;
+        }
+    }
+
+    enum glaciteCategory implements Category {
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/skyblock/dwarven/MiningLocationLabel.java:123]: GlaciteCategory

File: src/main/java/de/hysky/skyblocker/skyblock/dwarven/MiningLocationLabel.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        @Override
+        public int getColor() {
+            if (isTitanium) {
+                return 0xd8d6d8;
+            }
+            return 0x45bde0;
+        }
+    }
+
+    enum dwarvenEmissaries implements Category {
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/skyblock/dwarven/MiningLocationLabel.java:87]: DwarvenEmissaries

File: src/main/java/de/hysky/skyblocker/skyblock/dwarven/MiningLocationLabel.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+        RenderHelper.renderText(context, Text.literal(Math.round(distance) + "m").formatted(Formatting.YELLOW), posUp, scale, MinecraftClient.getInstance().textRenderer.fontHeight + 1, true);
+    }
+
+    interface Category {
+        String getName();
+
+        int getColor(); //all the color codes are the color of the block the waypoint is for
+    }
+
+    enum dwarvenCategory implements Category {
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/skyblock/dwarven/MiningLocationLabel.java:46]: DwarvenCategory

File: src/main/java/de/hysky/skyblocker/skyblock/dwarven/MiningLocationLabel.java [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.text.Text;
+import net.minecraft.util.Formatting;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Vec3d;
+
+public class MiningLocationLabel {
+    final Text name;
+    final Category category;
+    private final Vec3d centerPos;
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/skyblock/dwarven/MiningLocationLabel.java:15]: If all the fields are final you could convert this to a record

File: src/main/java/de/hysky/skyblocker/skyblock/dwarven/CommissionLabels.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+import net.minecraft.util.math.BlockPos;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class CommissionLabels {
+
+    private static final Map<String, MiningLocationLabel.dwarvenCategory> DWARVEN_LOCATIONS = Arrays.stream(MiningLocationLabel.dwarvenCategory.values()).collect(Collectors.toMap(MiningLocationLabel.dwarvenCategory::toString, Function.identity()));
+    private static final List<MiningLocationLabel.dwarvenEmissaries> DWARVEN_EMISSARYS = Arrays.stream(MiningLocationLabel.dwarvenEmissaries.values()).toList();
```
Comment (AzureAaron) [inline src/main/java/de/hysky/skyblocker/skyblock/dwarven/CommissionLabels.java:17]: DWARVEN_EMISSARIES

**Review by AzureAaron — DISMISSED** [review]

**Review by kevinthegreat1 — APPROVED** [review]
Works well. Although I think some options such as colored waypoints are kind of unnecessary.

**Pull request merged**